.github/workflows/create-release-tag.yml

```diff
@@ -0,0 +1,110 @@
+        id: get_tag
+        run: |
+          echo "old_tag_name=$(git ls-remote --tags origin | awk -F'/' '{print $3}' | grep -v '{}' | sort -V | tail -n1)" >> $GITHUB_OUTPUT
+      - name: print tag
+        id: print_tag
+        run: | 
+          echo "Old Tag=${{ steps.get_tag.outputs.old_tag_name }}"
+          echo "NEW_TAG=${{ github.event.inputs.tag_name }}" >> $GITHUB_ENV
+          echo "$(basename ${{ github.ref }})"
+
+      - name: Check if tag exists
+        id: check_tag
+        run: |
+         import sys
+         import subprocess
+         tag_name = "${{ github.event.inputs.tag_name }}"
+         command = ['git', 'tag', '-l', tag_name]
+         output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+         if output.decode() != "":
+           print(f"Error: Tag '{tag_name}' already exists.", file=sys.stderr)
+           sys.exit(1)
+         else:
+           print(f"Tag '{tag_name}' does not exists.")
+        
+        shell: python
+        continue-on-error: false
+
+#this works only if params.env contains image:tag_version_number
+  update-params-env:
+    runs-on: ubuntu-latest
+    needs: fetch-tag
+    outputs:
+      param_env: ${{ steps.read_params_env.outputs.params_env }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+           
+      - name: Read params.env
+        id: read_params_env
+        run: |
+          echo "params_env=$(cat config/overlays/odh/params.env)" >> $GITHUB_ENV
+              
+      - name: Update params.env with new release version
+        run: |     
+          sed -i 's|:v[0-9.]*\b|:${{ github.event.inputs.tag_name }}|gm' config/overlays/odh/params.env
+      - name: Commit changes
+        run: |
+          git config --global user.email "github-actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git add config/overlays/odh/params.env
+          git commit -m "Update image refs for odh release."
+          git push 
+
+      - name: Create Tag
+        id: create_tag
+        run: |
+          git tag -a ${{ github.event.inputs.tag_name }} -m "Prepare for ODH release ${{ github.event.inputs.tag_name }}"
+          git push origin ${{ github.event.inputs.tag_name }}
+
+  changelog:
+    name: Changelog
+    needs: [fetch-tag,update-params-env]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        with:
+          token: ${{ github.token }}
+          tag_name: ${{ github.event.inputs.tag_name }}
+          prerelease: false
+          draft: false
+    #this takes the path of payload to upload as an asset in the changelog
+          files: bin/*
+          generate_release_notes: true
+          name: ${{ github.event.inputs.tag_name }}
+       
```

Comment:
you might want to add a newline at the end of the file. viscose can be configured to do this for you automatically.